src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -967,6 +1003,23 @@ public void OrElse()
             tokens.AddLast(QueryOperatorToken.Or);
         }
 
+        [Obsolete("You shouldn't use this method directly. Use .Filter() only.")]
+        public void TurnOnFilter()
```

Comment:
This is for internal use, right? Can we mark it as `internal`? We shouldn't mark it as obsolete, it shouldn't be just available for users

Reply:
Unfortunately, we can't. We need to have this method in the interface for LinqProcessor.

Reply:
After direct talk we found a way to solve that. We'll do a similar trick as we do here:

https://github.com/ravendb/ravendb/blob/4a37bea09c16198d2d8f29d5fd79beb4c0f9f876/src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs#L2304-L2307

Reply:
Done